### ISSUE TITLE
[Feature] Send to queue with preset overrides.

### ISSUE THREAD

**Issue by koblue** [ArtVentureX/sd-webui-agent-scheduler#32]
tl;dr: Send image to queue with preset settings so you can send lowres images to img2img and queue them to be upscaled via the image browser as you go.

I have a ton of generations I want to upscale via the img2img tab, but the setup for each image is tedious especially as I'm using controlnet and the ultimate upscale script. My settings in this tab don't change each time but the input prompt, seed and image etc do.

Could there be a way to store certain settings as a preset and then use the prompt, image and seed etc to fill in the blanks then send that to the agent queue?

This may be out of scope for this extension but its the closest in function to what I am looking for. 

Also thank you for this extension. It's been incredibly useful.

**Comment by artventuredev** [comment]
@koblue I think this will be an interesting feature. Added to the backlog.

**Comment by CptGabok** [comment]
This would be amazing! And yes, thank you a ton for this extension. How can we best support you?

**Comment by CptGabok** [comment]
> tl;dr: Send image to queue with preset settings so you can send lowres images to img2img and queue them to be upscaled via the image browser as you go.

I found a way of manually hacking this functionality in. In your `sd-webui-agent-scheduler/agent_scheduler` folder, there is a `task_runner.py` file. Open that file, and in the `def parse_task_args` section (around line 186), beneath `script_args: list[Any] = parsed.get("script_args", [])`, you can add override settings.

To find the names of the settings you want to override, you can first add a line that will print to the Stable Diffusion console when you click Enqueue all the settings sent to Agent Schdeduler: `log.info(f"[AgentScheduler] named_args:{named_args}") # This will display in the console the setting names that are being sent to Agent Scheduler`

This is what my code looked like:
```
        script_args: list[Any] = parsed.get("script_args", []) # THIS ALREADY EXISTED, INCLUDED FOR ORIENTATION

        # Manually overriding specific settings
        # log.info(f"[AgentScheduler] named_args:{named_args}") # This will display in the console the setting names that are being sent to Agent Scheduler
        log.info(f"[AgentScheduler] Overriding settings")
        named_args['batch_size'] = 1
        named_args['enable_hr'] = True
        named_args['denoising_strength'] = 0.44
        named_args['hr_upscaler'] = '4x-UltraSharp'

        # Back to normal code

```

**Comment by CptGabok** [comment]
> ![image](https://user-images.githubusercontent.com/13137254/257777461-284d1649-8cca-42ba-9486-9eac1e536011.png)

Hey - not sure what this was meant to communicate. Apologies if I'm missing something.

Is there any movement on this feature? Or a way to simulate it using APIs (which is maybe what you're signaling here)?
